.github/workflows/test.yaml

```diff
@@ -10,125 +10,144 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
-    linting:
-        runs-on: ubuntu-latest
-
-        steps:
-        - uses: actions/checkout@v3
-
-        - name: Setup Python
-          uses: actions/setup-python@v4
-          with:
-              python-version: "3.10"
-
-        - name: Install Dependencies
-          run: |
-            python -m pip install --upgrade pip
-            pip install .[lint]
-
-        - name: Run Black
-          run: black --check .
-
-        - name: Run isort
-          run: isort --check-only .
-
-        - name: Run flake8
-          run: flake8 .
-
-        - name: Run mdformat
-          run: mdformat . --check
-
-    type-check:
-        runs-on: ubuntu-latest
-
-        steps:
-        - uses: actions/checkout@v3
-
-        - name: Setup Python
-          uses: actions/setup-python@v4
-          with:
-              python-version: "3.10"
-
-        - name: Install Dependencies
-          run: |
-            python -m pip install --upgrade pip
-            pip install .[lint,test]
-
-        - name: Run MyPy
-          run: mypy .
-
-    functional:
-        runs-on: ${{ matrix.os }}
-
-        strategy:
-            matrix:
-                os: [ubuntu-latest, macos-latest]   # eventually add `windows-latest`
-                python-version: [3.8, 3.9, "3.10", "3.11"]
-
-        env:
+  detect-changes:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Install changed-files action
+      run: npm install umani/changed-files
```

Comment:
is there a python version of this tool? That would fit better and not require installing npm